consumo servicio User

acgomez623 opened the pull request:
Falta la validación de la excepción de que el cliente no existe

acgomez623 committed:
consumo servicio User

Falta la validación de la excepción de que el cliente no existe

acgomez623 committed:
Correciones y test userClient

caferrerbeam commented on src/main/java/com/co/indra/coinmarketcap/watchlist/APIs/clients/UsersApiClient.java:
```diff
@@ -0,0 +1,40 @@
+                Users.class);
+
+        if (response.getStatusCode() != HttpStatus.OK) {
+            throw new BussinessException(ErrorCodes.WATCHLIST_USER_DOESNOT_EXISTS);
```
NOT_FOUND mejor...

caferrerbeam commented on src/main/java/com/co/indra/coinmarketcap/watchlist/controllers/WatchlistController.java:
```diff
@@ -21,9 +21,9 @@ public class WatchlistController {
-    public void create(@Valid @RequestBody Watchlist watchlist){
-        watchlistService.createWatchlist(watchlist);
+    @PostMapping(Routes.ID_USER_PATH)
+    public void create(@PathVariable String username, @Valid @RequestBody Watchlist watchlist){
```
validar el userName | userId que viene en el obj wathclist

acgomez623 committed:
Correcciones servicio User

Se valida el username desde el objeto watchlist, se cambia excepción por NOT FOUND y se corrigen sus pruebas respectivasd
